### PR TITLE
tt: split data directory

### LINKED ISSUE #326: Split the `lib` directory
Now the snapshots, WALs and vynil files are written to the same directory, but in reality a user may want to store the WALs files in one directory (on the one physical / logical storage) and snapshots to the other. And such functionality is actually used in production.
Needs to support such functionality in `tt` via `tt.yaml`. And in `tt init` when reading the `.tarantoolctl`.

### PR DESCRIPTION
**Pull request by psergee**
Current tt implementation uses common data directory for all artifacts: snapshots, WAL and vinyl logs. It is a common practive to place applications artifacts to different directories/partitions. tt config format is updated to support different locations for wal, memtx and vinyl artifacts.

Here is an example of the fs tree of the environment with separate wal, vinyl and snapshot dirs:
```
var/
├── lib
│   ├── memtx
│   │   └── myapp
│   │       └── s1-master
│   │           └── 00000000000000000000.snap
│   ├── vinyl
│   │   └── myapp
│   │       └── s1-master
│   │           ├── 00000000000000000000.vylog
│   │           └── 512
│   │               └── 0
│   └── wal
│       └── myapp
│           └── s1-master
│               └── 00000000000000000000.xlog
├── log
│   └── myapp
│       └── s1-master
│           └── s1-master.log
└── run
    └── myapp
        └── s1-master
            ├── s1-master.control
            └── s1-master.pid
```

Closes #326

Follow-up task for `pack` command is created: #328

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -106,7 +106,9 @@ file format:
         log_dir: path/to/log_dir
         bin_dir: path/to/bin_dir
         inc_dir: path/to/inc_dir
-        log_maxsize: num (MB)
```
Comment (LeonidVas) [inline README.rst:109]: Why did you decide to remove `log_maxsize`?

File: cli/init/print_tarantoolctl_cfg.lua [hunk 2]
```diff
@@ -10,23 +12,12 @@ if default_cfg == nil then
...
-        end
-    else
-        if data_dir ~= nil and lib_dir ~= data_dir then
-            io.stderr:write("Unable to identify data directory from taractoolctl config. " ..
-            "There is uncertainty between " .. lib_dir .. " and " .. data_dir .. "\n")
-            os.exit(1)
-        end
-    end
-end
+local d = {}
```
Comment (LeonidVas) [inline cli/init/print_tarantoolctl_cfg.lua:15]: Please do not use names that are one character long. They are inconvenient to use for future navigation and do not describe the object.

**Review by LeonidVas — APPROVED** [review]
Hi! Thank you for the patch.
Generally ok, see several comments bellow.

File: cli/init/print_tarantoolctl_cfg.lua (hunk 2 above)
Comment (psergee) [reply cli/init/print_tarantoolctl_cfg.lua]: Renamed.

File: README.rst (hunk 1 above)
Comment (psergee) [reply README.rst:109]: Accidental remove.

**Pull request merged**